lib/sidekiq/throttled/patches/super_fetch.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require "sidekiq"
+
+module Sidekiq
+  module Throttled
+    module Patches
+      module SuperFetch
+        # Retrieves job from redis.
+        #
+        # @return [Sidekiq::Throttled::UnitOfWork, nil]
+        def retrieve_work
+          work = super
+
+          if work && Throttled.throttled?(work.job)
+            Throttled.cooldown&.notify_throttled(work.queue)
+            requeue_throttled(work)
+            return nil
+          end
+
+          Throttled.cooldown&.notify_admitted(work.queue) if work
+
+          work
+        end
+
+        private
+
+        # Pushes job back to the head of the queue, so that job won't be tried
+        # immediately after it was requeued (in most cases).
+        #
+        # @note This is triggered when job is throttled. So it is same operation
+        #   Sidekiq performs upon `Sidekiq::Worker.perform_async` call.
+        #
+        # @return [void]
+        def requeue_throttled(work)
+          if work.respond_to?(:local_queue)
+            # if a SuperFetch UnitOfWork, SuperFetch will requeue it using lpush
+            work.requeue
+          else
+            # Fallback to BasicFetch behavior
+            redis { |conn| conn.lpush(work.queue, work.job) }
+          end
+        end
+
+        # Returns list of non-paused queues to try to fetch jobs from.
+        #
+        # @note It may return an empty array.
+        # @return [Array<Array(String, String)>]
+        def active_queues
```

Comment:
This looks the same as `Sidekiq::Throttled::Patches::BasicFetch.queues_cmd` but rewritted. Am I missing something? If I'm correct, is there a way we can reuse existing code?

Reply:
I believe SuperFetch is not inherited from BasicFetch: https://github.com/ixti/sidekiq-throttled/issues/111#issuecomment-1587436701

Reply:
Yup, SuperFetch does not inherit from BasicFetch, and SuperFetch does not implement `queues_cmd`